src/chainstate/stacks/miner.rs

```diff
@@ -971,40 +972,7 @@ impl StacksBlockBuilder {
         Ok(())
     }
 
-    /// Finish building the anchored block.
-    /// TODO: expand to deny mining a block whose anchored static checks fail (and allow the caller
-    /// to disable this, in order to test mining invalid blocks)
-    pub fn mine_anchored_block(&mut self, clarity_tx: &mut ClarityTx) -> StacksBlock {
-        assert!(!self.anchored_done);
-
-        // add miner payments
-        if let Some((ref miner_reward, ref user_rewards, ref parent_reward)) = self.miner_payouts {
-            // grant in order by miner, then users
-            let matured_ustx = StacksChainState::process_matured_miner_rewards(
-                clarity_tx,
-                miner_reward,
-                user_rewards,
-                parent_reward,
-            )
-            .expect("FATAL: failed to process miner rewards");
-
-            clarity_tx.increment_ustx_liquid_supply(matured_ustx);
-        }
-
-        // process unlocks
-        let (new_unlocked_ustx, _) =
-            StacksChainState::process_stx_unlocks(clarity_tx).expect("FATAL: failed to unlock STX");
-
-        clarity_tx.increment_ustx_liquid_supply(new_unlocked_ustx);
-
-        // mark microblock public key as used
-        StacksChainState::insert_microblock_pubkey_hash(
-            clarity_tx,
-            self.header.total_work.work as u32,
-            &self.header.microblock_pubkey_hash,
-        )
-        .expect("FATAL: failed to insert microblock pubkey hash");
-
+    pub fn set_root_hash_and_get_block(&mut self, clarity_tx: &mut ClarityTx) -> StacksBlock {
```

Comment:
Can you call this something less dependent on the implementation?  Maybe something like `finalize_block()` or something?